### PR TITLE
[Identity] Update documentation

### LINKED ISSUE #31064: Azure Identity October Release Doc Consistency Review
tracking item for README, Samples, TROUBLESHOOT, and Changelog consistency review

### PR DESCRIPTION
**Pull request by pvaneck**
This updates the README and TROUBLESHOOTING files for greater consistency with other language docs and for general enhancement.

Closes: #31064

### CODE REVIEW

File: sdk/identity/Azure.Identity/README.md
Comment (jsquire) [inline sdk/identity/Azure.Identity/README.md:167]:
```suggestion
[AzureAuthorityHosts](https://learn.microsoft.com/dotnet/api/azure.identity.azureauthorityhosts?view=azure-dotnet)
```

File: sdk/identity/Azure.Identity/README.md
Comment (jsquire) [inline sdk/identity/Azure.Identity/README.md:169]: We'll need to make these snippets so that the code is compiled, and smoke tested to avoid getting stale.   Please see: [Updating Sample Snippets](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets).

**Review by jsquire — DISMISSED** [review]

File: sdk/identity/Azure.Identity/README.md
Comment (scottaddie) [inline sdk/identity/Azure.Identity/README.md:270]:
```suggestion
For more information on dealing with errors arising from failed requests to Azure AD or managed identity endpoints, see the Azure AD [documentation on authorization error codes][aad_err_doc].
```

File: sdk/identity/Azure.Identity/TROUBLESHOOTING.md
Comment (scottaddie) [inline sdk/identity/Azure.Identity/TROUBLESHOOTING.md:9]:
```suggestion
- [Permission issues](#permission-issues)
```

Comment (pvaneck) [reply sdk/identity/Azure.Identity/TROUBLESHOOTING.md:9]: I think it makes sense to keep this as a sub-item under "Handle Azure Identity exceptions" since it still deals with exceptions (Python and Java also have this as a sub-item).

Comment (scottaddie) [reply sdk/identity/Azure.Identity/TROUBLESHOOTING.md:9]: Disregard my suggestion. I agree.

File: sdk/identity/Azure.Identity/README.md
Comment (scottaddie) [inline sdk/identity/Azure.Identity/README.md:5]:
```suggestion
  [Source code][source] | [Package (NuGet)][package] | [API reference documentation][identity_api_docs] | [Azure AD documentation][aad_doc]
```

File: sdk/identity/Azure.Identity/README.md
Comment (scottaddie) [inline sdk/identity/Azure.Identity/README.md:29]:
```suggestion
To authenticate in Visual Studio, select the **Tools** > **Options** menu to launch the Options dialog. Then navigate to the `Azure Service Authentication` options to sign in with your Azure AD account.
```

File: sdk/identity/Azure.Identity/README.md
Comment (scottaddie) [inline sdk/identity/Azure.Identity/README.md:168]:
```suggestion
Credentials default to authenticating to the Azure AD endpoint for the Azure public cloud. To access resources in other clouds, such as Azure Government or a private cloud, configure credentials with the `authority` argument. [AzureAuthorityHosts](https://learn.microsoft.com/dotnet/api/azure.identity.azureauthorityhosts?view=azure-dotnet) defines authorities for well-known clouds:
```

File: sdk/identity/Azure.Identity/README.md
Comment (scottaddie) [inline sdk/identity/Azure.Identity/README.md:175]:
```suggestion

Not all credentials require this configuration. Credentials which authenticate through a development tool, such as `AzureCliCredential`, use that tool's configuration.
```

File: sdk/identity/Azure.Identity/README.md
Comment (scottaddie) [inline sdk/identity/Azure.Identity/README.md:334]:
```suggestion
This library doesn't currently support scenarios relating to the [Azure AD B2C](https://learn.microsoft.com/azure/active-directory-b2c/overview) service.
```

File: sdk/identity/Azure.Identity/README.md
Comment (scottaddie) [inline sdk/identity/Azure.Identity/README.md:336]:
```suggestion
Open issues for the `Azure.Identity` library can be found [here](https://github.com/Azure/azure-sdk-for-net/issues?q=is%3Aissue+is%3Aopen+label%3AAzure.Identity).
```

File: sdk/identity/Azure.Identity/TROUBLESHOOTING.md
Comment (scottaddie) [inline sdk/identity/Azure.Identity/TROUBLESHOOTING.md:244]:
```suggestion
You can manually verify that Azure PowerShell is properly authenticated, and can obtain tokens. First, use the `Get-AzContext` command to verify the account which is currently logged in to the Azure CLI.
```

File: sdk/identity/Azure.Identity/TROUBLESHOOTING.md
Comment (scottaddie) [inline sdk/identity/Azure.Identity/TROUBLESHOOTING.md:254]:
```suggestion
Once you've verified Azure PowerShell is using correct account, validate that it's able to obtain tokens for this account:
```

**Review by scottaddie — APPROVED** [review]
One final round of suggestions, and then this PR looks good to me.

**Comment by pvaneck** [comment]
@schaabs Would like to get your review/approval before merging.

**Review by schaabs — APPROVED** [review]

**Comment by pvaneck** [comment]
Thanks all for the reviews. :)

**Pull request merged**
